wp_api/src/request/endpoint.rs

```diff
@@ -80,6 +80,39 @@ impl ApiBaseUrl {
         site_base_url.try_into()
     }
 
+    pub(crate) fn derived_wp_login_url(&self) -> Url {
+        let mut url = self.url.clone();
+
+        if let Some(segments) = url.path_segments() {
+            if segments.last() == Some("") {
+                url.path_segments_mut()
+                    .expect("ApiBaseUrl is a full HTTP URL")
+                    .pop();
+            }
+        }
```

Comment:
> I can't remember if/when the segment could be an empty string

It'd be empty if there is trailing `/`, like `https://example.com/f/`

> Could you add a test case that is handled by this implementation?

So, I kinda cheated here. I didn't add new unit tests to cover this code block because they would fail. The cause is not this new code though. It's because `ApiBaseUrl` can't parse some URLs with trailing slash. I have fixed the issue in https://github.com/Automattic/wordpress-rs/pull/353. Do you mind having a look at that PR first?

Reply:
I have added more test cases in https://github.com/Automattic/wordpress-rs/pull/327/commits/c30f3fb967c5e2d58e55a9f2a8caa9c87f52b407